Composer/packages/types/src/shell.ts

```diff
@@ -23,9 +23,9 @@ import { FeatureFlagKey } from './featureFlags';
 import { TelemetryClient } from './telemetry';
 
 /** Recursively marks all properties as optional. */
-type AllPartial<T> = {
-  [P in keyof T]?: T[P] extends (infer U)[] ? AllPartial<U>[] : T[P] extends object ? AllPartial<T[P]> : T[P];
-};
+// type AllPartial<T> = {
```

Comment:
@a-b-r-o-w-n I had some type troubles with user settings relying on Partial in some places and AllPartial in others.

Reply:
Can you elaborate on what troubles you were facing?

Reply:
TypeScript was not happy with it